### DISCUSSION TITLE
Welcome to jsrepo Discussions!

### DISCUSSION THREAD

**Discussion by ieedan** [jsrepojs/jsrepo#210] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by janzheng** [comment]
Hello! 

Love this idea and looking forward to using it! One huge aspect to me is the ability to tear down / change published projects, and to be able to have an "npm or jsr-like scratchpad without the permanence". Thanks for making this!

I'd love to have a "local registry" first where I can mess around with blocks and how they connect without uploading/downloading to a registry — this would make it easier to reference code changes or even data repos (e.g. large CSVs/JSON/parquet files) and make them easily importable into various projects

**Reply by ieedan** [reply]
> Hello! 
> 
> 
> 
> Love this idea and looking forward to using it! One huge aspect to me is the ability to tear down / change published projects, and to be able to have an "npm or jsr-like scratchpad without the permanence". Thanks for making this!
> 
> 
> 
> I'd love to have a "local registry" first where I can mess around with blocks and how they connect without uploading/downloading to a registry — this would make it easier to reference code changes or even data repos (e.g. large CSVs/JSON/parquet files) and make them easily importable into various projects

I have thought about local registries for a while and they come with their own challenges but are probably possible.

We also support private repositories on GitHub so that you don't have to make that data public!

**Reply by janzheng** [reply]
nice, I saw that! I also would feel bad for uploading hundreds of mbs of data to your repo and using it as a sort of CDN for my junk :P 
I understand about challenges, and I just realized that there isn't much of a "local reference manager" where you can just import "@/local-dir/file" across projects. That's probably outside of jsrepo's scope, but that'd be so useful. Currently I just use symlinks but they also come with their own headaches

**Reply by ieedan** [reply]
> nice, I saw that! I also would feel bad for uploading hundreds of mbs of data to your repo and using it as a sort of CDN for my junk :P 
> 
> I understand about challenges, and I just realized that there isn't much of a "local reference manager" where you can just import "@/local-dir/file" across projects. That's probably outside of jsrepo's scope, but that'd be so useful. Currently I just use symlinks but they also come with their own headaches

Don't feel bad it's on Microsoft's dime!

**Reply by ieedan** [reply]
> nice, I saw that! I also would feel bad for uploading hundreds of mbs of data to your repo and using it as a sort of CDN for my junk :P 
> 
> I understand about challenges, and I just realized that there isn't much of a "local reference manager" where you can just import "@/local-dir/file" across projects. That's probably outside of jsrepo's scope, but that'd be so useful. Currently I just use symlinks but they also come with their own headaches

Also yeah even if I implement local directories it's going to copy that massive file all around your system so I think symlinks would probably be the best bet there!

**Reply by janzheng** [reply]
ah that makes sense.

again, out of scope probably, but it'd be neat to have a symlink/cache system where files and registry is copied once into a cache, like pnpm/deno, and then can be referenced into as many projects as you want, with semantic versioning etc. and essentially acts as a symlink, but can be integrated w/ a build step to copy the contents over on build

**Reply by ieedan** [reply]
Yeah I think that's a bit out of scope it is meant to work more like something like `shadcn/ui` but for everyone.

**Reply by ieedan** [reply]
While it isn't really a symlink solution I am adding support for self-hosted registries in https://github.com/ieedan/jsrepo/pull/351

This would allow you to test a registry locally by serving it's files on something like [LiveServer](https://marketplace.visualstudio.com/items?itemName=ritwickdey.LiveServer) and then accessing them like:
```bash
jsrepo add http://localhost:5500/utils/add
```

**Comment by GiorgioDots** [comment]
Hello!

I found your project on daily.dev early this week and I looked forward to try it today. It's really easy to setup the registry and use it in a project, so thank you for creating this powerful tool!

While I was waiting to try it I had a few questions: 
- Are you planning to make angular compatible (html and scss files)? I think it will be a big improvements for angular ui libraries. It currently doesn't have a shadcn/ui-like library, and I think it would benefit a lot thanks to your project. 
- I saw from jsrepo documentation and confirmed by looking at the code, that Azure Devops is not currently supported, did you consider to make this git repository compatible?

I would love to contribute and add those features and help your project grow further, is it possible to do it?

Thank you again! 🖖🏻

**Reply by ieedan** [reply]
> Hello!
> 
> I found your project on daily.dev early this week and I looked forward to try it today. It's really easy to setup the registry and use it in a project, so thank you for creating this powerful tool!
> 
> While I was waiting to try it I had a few questions: 
> - Are you planning to make angular compatible (html and scss files)? I think it will be a big improvements for angular ui libraries. It currently doesn't have a shadcn/ui-like library, and I think it would benefit a lot thanks to your project. 
> - I saw from jsrepo documentation and confirmed by looking at the code, that Azure Devops is not currently supported, did you consider to make this git repository compatible?
> 
> I would love to contribute and add those features and help your project grow further, is it possible to do it?
> 
> Thank you again! 🖖🏻

I personally don't know much about angular or Azure devops luckily things are setup in a way that make it pretty easy to add these things. 

If you would like to give it a shot please do! If not just open an issue to track each of those things and I can try and work on it when I have the time!

**Reply by ieedan** [reply]
Is Azure Devops a payed only product?

Also reading on angular it seems like we don't really need to do any new dependency resolution? Just need to add the ability to use html and scss?

**Reply by GiorgioDots** [reply]
Azure DevOps has a free tier with unlimited repositories and up to 5 users per organization, but you need an Azure account to access DevOps and i think it require a credit card to sign up.
Today I looked the [API docs](https://learn.microsoft.com/en-us/rest/api/azure/devops/build/source-providers?view=azure-devops-rest-7.1l) to retrieve the source code of a repository and it seems not very difficult to integrate into your project. But I will dig further in the next days to check if I'm missing something.

For angular, it needs the support for html and scss, it should also exclude automatically any "@angular/*" dependency from any *.ts file. I'll do some more tests and research for this feature too in the next days.

Thanks!

**Reply by ieedan** [reply]
HTML and Sass support in #308

**Reply by GiorgioDots** [reply]
> HTML and Sass support in #308

Thank you so much! 🙇‍♂️ I'll try it in a moment
